sys/trickle/trickle.c

```diff
@@ -53,27 +53,27 @@ void trickle_interval(trickle_t *trickle)
     trickle->c = 0;
     trickle->t = (trickle->I / 2) + (rand() % ((trickle->I / 2) + 1));
 
-    vtimer_remove(&trickle->msg_callback_timer);
-    trickle->msg_callback_time = timex_set(0, trickle->t * 1000);
-    vtimer_set_msg(&trickle->msg_callback_timer, trickle->msg_callback_time, trickle->pid,
-            trickle->callback_msg_type, trickle);
-
-    vtimer_remove(&trickle->msg_interval_timer);
-    trickle->msg_interval_time = timex_set(0, trickle->I * 1000);
-    vtimer_set_msg(&trickle->msg_interval_timer, trickle->msg_interval_time, trickle->pid,
-            trickle->interval_msg_type, trickle);
+    xtimer_remove(&trickle->msg_callback_timer);
```

Comment:
This is only needed if it might lead to a race when modifying the timer's underlying msg structure and the timer might fire in between. Otherwise, xtimer always makes sure it doesn't break when adding an already set timer (calling _remove() if necessary)